First implementation of method extract_item:

AntonyColapietro opened the pull request:
First implementation of method extract_item:
Expects three parameters: unlocalized name of the item, number of the items to transfer, side of which the peripheral must transfer the item (acceptable values are : "north","south","up","down","est","west").
Moves specified items from the network to the inventory at the specified side of the peripheral.
Name of the items follow the same format as those returned by the method "list_items"

Commit:
First implementation of method extract_item:
Expects three parameters: unlocalized name of the item, number of the items to transfer, side of which the peripheral must transfer the item.
Moves specified items from the network to the inventory at the specified side of the peripheral.
Name of the items follow the same format as those returned by the method list_items

admiral0 commented on src/main/java/it/antani/cc/impl/refinedstorage/ControllerImpl.java:
```diff
@@ -43,4 +48,99 @@
+                    .findFirst();
+        }
+        else
+            return new Object[]{null,"First parameter must be the itemstack name you need to retrieve"};
```
Please throw errors instead of successfully returning. The lua internal runtime exposes that as errors

admiral0 commented on src/main/java/it/antani/cc/impl/refinedstorage/ControllerImpl.java:
```diff
@@ -43,4 +48,99 @@
+
+
+        //Check the entity on the specified side
+        TileEntity targetEntity = controller.getNode().getWorld().getTileEntity(controller.getNode().getPosition().offset(facing));
```
Is this the side of the computer or the target tile entity?

admiral0 commented on src/main/java/it/antani/cc/impl/refinedstorage/ControllerImpl.java:
```diff
@@ -43,4 +48,99 @@
+     *  Expected args passed from the Lua interpreter are:
+     *  String itemstackName : itemstack name to transfer
+     *  Double numberOfItems : number of items to transfer
+     *  String side : side where to transfer the item (north,south,up,down,est,west)
```
s/est/east/

admiral0 commented on src/main/java/it/antani/cc/impl/refinedstorage/ControllerImpl.java:
```diff
@@ -43,4 +48,99 @@
+        //Put the itemstack in the destination inventory
+        ItemHandlerHelper.insertItemStacked(handler, extracted, false);
+        //Remove the itemstack of specified quantity from the network as it has been transferred
+        controller.getNode().getItemStorageCache().getList().remove(stack.get(),transferableAmount);
```
why is that necessary?

admiral0 reviewed (changes requested):
Needs rework on error handling

Commit:
Second iteration for implementation of method extract_item:
Expects up to three parameters: 
- (Mandatory) Unlocalized name of the item, 
- (Mandatory) Number of the items to transfer,
- (Optional for turtles, mandatory for computers) Side of the computer/turtle exposed to an inventory
Moves specified items from the network to the inventory at the specified side of the computer/turtle when third parameter is specified, move them to the current selected slot of the turtle otherwise.

Name of the items follow the same format as those returned by the method list_items.

admiral0 commented on src/main/java/it/antani/cc/impl/refinedstorage/ControllerImpl.java:
```diff
@@ -76,49 +85,134 @@
 
-        //Check the entity on the specified side
-        TileEntity targetEntity = controller.getNode().getWorld().getTileEntity(controller.getNode().getPosition().offset(facing));
+        for (EnumFacing enumFacing : EnumFacing.values()) {
```
there is a better way....
